subsys/bluetooth/shell/gatt.c

```diff
@@ -950,6 +950,70 @@ static int cmd_notify(const struct shell *sh, size_t argc, char *argv[])
 	return 0;
 }
 
+#if defined(CONFIG_BT_GATT_NOTIFY_MULTIPLE)
+static int cmd_notify_mult(const struct shell *sh, size_t argc, char *argv[])
+{
+	const size_t max_cnt = CONFIG_BT_L2CAP_TX_BUF_COUNT;
+	struct bt_gatt_notify_params params[max_cnt];
+	const size_t min_cnt = 1U;
+	unsigned long data;
+	unsigned long cnt;
+	uint16_t cnt_u16;
+	int err = 0;
+
+	if (!echo_enabled) {
+		shell_error(sh, "No clients have enabled notifications for the vnd1_echo CCC.");
+
+		return -ENOEXEC;
+	}
+
+	cnt = shell_strtoul(argv[1], 10, &err);
```

Comment:
FYI, `shell_strtoul` leaves `err` untouched on success. There is no error in the code as it is, because all errors go directly to an early return. But it may be good practice to reset `err` to `0` before each error catch grouping (i.e. above this line in this case), in case someone adds a change that sets `err` above this line. I'm not too worried about this though, so I'm approving.

Reply:
Right. 

Is it by design that `err` is not set to `0` on success, or is it an oversight?